src/app/core/store/restore/restore.effects.ts

```diff
@@ -93,13 +106,55 @@ export class RestoreEffects {
     whenTruthy(),
     take(1),
     switchMap(() =>
-      interval(1000).pipe(
+      interval(300).pipe(
         takeWhile(() => isPlatformBrowser(this.platformId)),
         withLatestFrom(this.store$.pipe(select(getLoggedInUser)), this.cookieService.cookieLawSeen$),
         filter(([, , cookieLawAccepted]) => cookieLawAccepted),
         map(([, user]) => ({ user, apiToken: this.cookieService.get('apiToken') })),
         filter(({ user, apiToken }) => user && !apiToken),
-        mapTo(new LogoutUser())
+        concatMapTo([
+          new LogoutUser(),
+          new WarningMessage({
+            message: 'session.timeout.notification.message',
+            duration: 0,
+            title: 'session.timeout.notification.title',
+          }),
+        ])
+      )
+    )
+  );
+
+  @Effect()
+  removeAnonymousBasketIfTokenVanishes$ = this.appRef.isStable.pipe(
+    whenTruthy(),
+    take(1),
+    switchMap(() =>
+      interval(300).pipe(
+        takeWhile(() => isPlatformBrowser(this.platformId)),
+        withLatestFrom(
+          this.store$.pipe(select(getLoggedInUser)),
+          this.store$.pipe(select(getCurrentBasket)),
+          this.cookieService.cookieLawSeen$
+        ),
+        filter(([, , , cookieLawAccepted]) => cookieLawAccepted),
+        map(([, user, basket]) => ({ user, basket, apiToken: this.cookieService.get('apiToken') })),
+        filter(({ user, basket, apiToken }) => !user && basket && !apiToken),
+        mapTo(new ResetBasket())
+      )
+    )
+  );
+
+  @Effect()
+  sessionKeepAlive$ = this.appRef.isStable.pipe(
+    takeWhile(() => isPlatformBrowser(this.platformId)),
+    whenTruthy(),
+    take(1),
+    switchMap(() =>
+      this.store$.pipe(
+        select(getCurrentBasket),
+        switchMap(basket =>
+          interval(RestoreEffects.SESSION_KEEP_ALIVE).pipe(mergeMapTo(basket ? [new LoadBasket()] : []))
```

Comment:
Will this work with the design view that uses zone stability as well?

Reply:
In theory this should work because the first isStable is being waited on.

Reply:
Sure, but only at startup. Am I right assuming this won't work after route changes? The interval is still active then.

Reply:
yes, but this was not the initial requirement. SFE Adapter just needs the first isStable to come through for initialization. Cypress Tests make sure this is always happening.

Reply:
I still have concerns with this. SFE adapter uses isStable on every route change ([see here](https://github.com/intershop/intershop-pwa/blob/develop/src/app/shared/cms/sfe-adapter/sfe-adapter.service.ts#L139).
With a long-running interval, this will work for the first route, but never again after that.

Reply:
good point. Including check for `SfeAdapterService.isInitialized()`